File: app/src/main/java/com/example/feel_journal/data/model/Category.kt
package com.example.feel_journal.data.model

import com.google.gson.annotations.SerializedName

data class Category(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String,
    @SerializedName("type") val type: CategoryType,
    @SerializedName("createdAt") val createdAt: String
)

File: server/Models/Category.cs
using System.ComponentModel.DataAnnotations;

namespace FeelJournal.Models;

public class Category
{
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(255)]
    public string Name { get; set; } = string.Empty;

    [Required]
    public CategoryType Type { get; set; } = CategoryType.Neutral;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<EmotionEntry> EmotionEntries { get; set; } = new List<EmotionEntry>();
}
